air/src/stack/mod.rs

```diff
@@ -1,3 +1,100 @@
-use super::EvaluationFrame;
+use super::{EvaluationFrame, Felt, FieldElement, TransitionConstraintDegree, Vec};
+use vm_core::{FMP_COL_IDX, STACK_TRACE_OFFSET};
 
+mod field_ops;
 pub mod op_flags;
+mod stack_manipulation;
+mod system_ops;
+
+// STACK OPERATIONS TRANSITION CONSTRAINTS
+// ================================================================================================
+
+/// Build the transition constraint degrees for the stack module and all the stack operations.
+pub fn get_transition_constraint_degrees() -> Vec<TransitionConstraintDegree> {
+    // system operations contraints degrees.
+    let mut degrees = system_ops::get_transition_constraint_degrees();
+    // field operations contraints degrees.
+    degrees.append(&mut field_ops::get_transition_constraint_degrees());
+    // stack manipulation operations contraints degrees.
+    degrees.append(&mut stack_manipulation::get_transition_constraint_degrees());
+
+    degrees
+}
+
+/// Returns the number of transition constraints for the stack operations.
+pub fn get_transition_constraint_count() -> usize {
+    system_ops::get_transition_constraint_count()
+        + field_ops::get_transition_constraint_count()
+        + stack_manipulation::get_transition_constraint_count()
+}
+
+/// Enforces constraints for the stack module and all stack operations.
+pub fn enforce_constraints<E: FieldElement<BaseField = Felt>>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+) -> usize {
+    let mut index = 0;
+
+    let op_flag = op_flags::OpFlags::new(frame);
+
+    // Enforces stack operations unique constraints.
+    index += enforce_unique_constraints(frame, result, &op_flag);
+
+    index
+}
+
+/// Enforces unique constraints for all the stack ops.
+pub fn enforce_unique_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: &op_flags::OpFlags<E>,
+) -> usize {
+    let mut constraint_offset = 0;
+
+    // system operations transition constraints.
+    system_ops::enforce_constraints(frame, result, op_flag);
+
+    constraint_offset += system_ops::get_transition_constraint_count();
+
+    // field operations transition constraints.
+    field_ops::enforce_constraints(frame, &mut result[constraint_offset..], op_flag);
+
+    constraint_offset += field_ops::get_transition_constraint_count();
+
+    // stack manipulation operations transition constraints.
+    stack_manipulation::enforce_constraints(frame, &mut result[constraint_offset..], op_flag);
+
+    constraint_offset += stack_manipulation::get_transition_constraint_count();
+
+    constraint_offset
+}
+
+trait EvaluationFrameExt<E: FieldElement> {
+    // --- Column accessors -----------------------------------------------------------------------
+
+    /// Returns the current value of the specified index in the stack. It assumes that the index is
+    /// a valid index.
```

Comment:
nit: I'd say "Returns the current value at the specified index in the stack. ..."